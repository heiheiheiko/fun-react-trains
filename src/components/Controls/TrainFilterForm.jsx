import React from 'react';
import { connect } from "react-redux";
import PropTypes from 'prop-types';
import { updateTrainFilter } from "../../redux/actions";

class TrainFilterForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      filterFieldValue: "",
    };
  }

  onChange(){
    this.setState({filterFieldValue: event.target.value});
  }

  onSubmit(){
    event.preventDefault();
    this.props.updateTrainFilter(this.state.filterFieldValue);
  }

  render() {
    return (
      <form onSubmit={() => this.onSubmit()}>
        <div className="field has-addons">
          <div className="control">
            <input 
              className="input"
              type="text" 
              name="filter" 
              value={this.state.filterFieldValue} 
              placeholder="filter by train number..."
              onChange={() => this.onChange()} 
            />
          </div>
          <div className="control">
            <input className="button is-primary" type="submit" value="apply" />
          </div>
        </div>
      </form>
    );
  }
}

TrainFilterForm.propTypes = {
  updateTrainFilter: PropTypes.func,
};

const mapDispatch = dispatch => ({
  updateTrainFilter: trainFilterString => dispatch(updateTrainFilter(trainFilterString)) 
});
const ConnectedComponent = connect(null, mapDispatch)(TrainFilterForm);
export default ConnectedComponent;