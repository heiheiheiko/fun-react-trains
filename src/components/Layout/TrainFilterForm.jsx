import React from 'react';
import { connect } from "react-redux";
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl'
import { updateTrainFilter } from "../../redux/actions";

class TrainFilterForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      filterFieldValue: "",
    };
  }

  onChange(){
    this.setState({filterFieldValue: event.target.value});
  }

  onSubmit(){
    event.preventDefault();
    this.props.updateTrainFilter(this.state.filterFieldValue);
  }

  render() {
    return (
      <form onSubmit={() => this.onSubmit()}>
        <div className="field has-addons">
          <div className="control">
            <FormattedMessage id="aside.trainFilterForm.placeholder"> 
              {(trans) => (
                <input 
                  className="input"
                  type="text" 
                  name="filter" 
                  value={this.state.filterFieldValue} 
                  placeholder={trans}
                  onChange={() => this.onChange()} 
                />
              )}
            </FormattedMessage>
          </div>
          <div className="control">
            <FormattedMessage id="actions.apply"> 
              {(trans) => (  
                <input className="button is-primary" type="submit" value={trans} />
              )}
            </FormattedMessage>
          </div>
        </div>
      </form>
    );
  }
}

TrainFilterForm.propTypes = {
  updateTrainFilter: PropTypes.func,
};

const mapDispatch = dispatch => ({
  updateTrainFilter: trainFilterString => dispatch(updateTrainFilter(trainFilterString)) 
});
const ConnectedComponent = connect(null, mapDispatch)(TrainFilterForm);
export default ConnectedComponent;
