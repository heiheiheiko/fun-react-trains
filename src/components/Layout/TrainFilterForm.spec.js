import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import SpecContainer from '../../../__specs__/helpers/SpecReduxIntelContainer';

import TrainFilterForm from './TrainFilterForm';

describe('<TrainFilterForm>', () => {
  const initialState = {};
  const mockStore = configureStore();
  let store, componentContainer;

  beforeEach(() => {
    store = mockStore(initialState)
    componentContainer = <SpecContainer store={store}> <TrainFilterForm /> </SpecContainer>;
  })

  it('could be renderd', () => {
    expect(mount(componentContainer).find(TrainFilterForm).length).toEqual(1)
  });

  it('matched the snapshot', () => {
    const renderedValue =  renderer.create(componentContainer).toJSON()
    expect(renderedValue).toMatchSnapshot();
  });
});