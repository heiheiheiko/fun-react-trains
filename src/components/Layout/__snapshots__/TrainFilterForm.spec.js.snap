// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrainFilterForm> matched the snapshot 1`] = `
<div>
   
   
  <form
    onSubmit={[Function]}
  >
    <div
      className="field has-addons"
    >
      <div
        className="control"
      >
        <input
          className="input"
          name="filter"
          onChange={[Function]}
          placeholder="filter by train number..."
          type="text"
          value=""
        />
      </div>
      <div
        className="control"
      >
        <input
          className="button is-primary"
          type="submit"
          value="apply"
        />
      </div>
    </div>
  </form>
   
   
</div>
`;
